src/clj_gatling/simulation.clj

```diff
@@ -82,39 +82,53 @@
     :else
     [nil context [nil nil]]))
 
-(defn- continue-run? [{:keys [runner sent-requests simulation-start next-run-at force-stop-ch]}]
-  (if (poll! force-stop-ch)
-    (do
-      (println "Force stop requested. Not starting new scenarios anymore")
-      false)
-    (runners/continue-run? runner sent-requests simulation-start next-run-at)))
+(defn- continue-fn [{:keys [runner simulation-start force-stop]}]
+  (fn [sent-requests next-run]
+    (if @force-stop
+      false
+      (do (swap! simulation-start #(or % (LocalDateTime/now)))
+          (runners/continue-run? runner sent-requests @simulation-start next-run)))))
```

Comment:
I coverted `force-stop-ch` to an atom because I was getting issues with it working sometimes where there were multiple runners who all polled the channel. It seemed like eventually some of them stopped getting the stop command from it.